### PR TITLE
add isDataTrieMigrated func on proxy

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: blockchain/endpointProviders/baseEndpointProvider.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ const (
 	rawStartOfEpochValidators  = "internal/json/startofepoch/validators/by-epoch/%d"
 	esdt                       = "address/%s/esdt/%s"
 	nft                        = "address/%s/nft/%s/nonce/%d"
+	isDataTrieMigrated        = "address/%s/is-data-trie-migrated"
```
Comment (iulianpascalau) [inline blockchain/endpointProviders/baseEndpointProvider.go:25]: indent problem? Missing one whitespace before `=` ?

File: blockchain/endpointProviders/baseEndpointProvider.go [hunk 2]
```diff
@@ -120,3 +121,7 @@ func (base *baseEndpointProvider) GetRawStartOfEpochMetaBlock(epoch uint32) stri
 func (base *baseEndpointProvider) GetValidatorsInfo(epoch uint32) string {
 	return fmt.Sprintf(rawStartOfEpochValidators, epoch)
 }
+
+func (base *baseEndpointProvider) IsDataTrieMigrated(addressAsBech32 string) string {
```
Comment (iulianpascalau) [inline blockchain/endpointProviders/baseEndpointProvider.go:125]: missing comment

File: blockchain/proxy.go [hunk 3]
```diff
@@ -648,6 +648,35 @@ func (ep *proxy) GetNFTTokenData(
 	return response.Data.TokenData, nil
 }
 
+// IsDataTrieMigrated returns true if the data trie of the given account is migrated
+func (ep *proxy) IsDataTrieMigrated(ctx context.Context, address sdkCore.AddressHandler) (bool, error) {
```
Comment (iulianpascalau) [inline blockchain/proxy.go:652]: unit tests?

File: blockchain/proxy.go [hunk 4]
```diff
@@ -648,6 +648,35 @@ func (ep *proxy) GetNFTTokenData(
...
+	if err != nil {
+		return false, err
+	}
+	if response.Error != "" {
+		return false, errors.New(response.Error)
+	}
+
+	isMigrated, ok := response.Data["isMigrated"]
+	if !ok {
+		return false, errors.New("invalid response")
```
Comment (ssd04) [inline blockchain/proxy.go:674]: more elaborated error message?

File: blockchain/endpointProviders/baseEndpointProvider.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply blockchain/endpointProviders/baseEndpointProvider.go:25]: Added.

File: blockchain/endpointProviders/baseEndpointProvider.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply blockchain/endpointProviders/baseEndpointProvider.go:125]: Added.

File: blockchain/proxy.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply blockchain/proxy.go:652]: Added.

File: blockchain/proxy.go (hunk 4 above)
Comment (BeniaminDrasovean) [reply blockchain/proxy.go:674]: Done.

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**